pkg/api/router.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2021 The TestGrid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"context"
+	"path"
+
+	v1 "github.com/GoogleCloudPlatform/testgrid/pkg/api/v1"
+	"github.com/GoogleCloudPlatform/testgrid/util/gcs"
+	"github.com/gorilla/mux"
```

Comment:
What is this providing over the built-in muxing packages?

Reply:
Subrouters aren't supported (to my knowledge) in the built-in Router objects: https://github.com/GoogleCloudPlatform/testgrid/pull/490/files#diff-ce9544cf2f046068d8443822bbe9e602876c44adb5e55e68381ff1ce4e69283cR44-R45

It's a pretty minor quality-of-life change, though; this could be architected to not use the gorilla/mux package. Do you think that would be better?

Reply:
In general I think N deps is easier to reason about and maintain than N+1 deps. So the cost is non-zero, but the benefits can outweigh those costs.

Probably fine here, not totally clear on why this is necessary/helpful.